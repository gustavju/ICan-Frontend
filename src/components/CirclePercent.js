import React from 'react';

const CirclePercent = ({ percent, percentColor }) => {
    let percentNum = parseInt(percent);
    // 90 is from svg r (radius)
    let strokeDashoffset = ((100 - percentNum) / 100) * (Math.PI * (90 * 2));
    let stroke = percentColor;
    return (
        <div className="circle-percent__label" data-pct={percent}>
            <svg class="circle-percent__svg" width="200" height="200" viewport="0 0 100 100" version="1.1" xmlns="http://www.w3.org/2000/svg">
                <circle r="90" cx="100" cy="100" strokeDasharray="565.48" strokeDashoffset="0"></circle>
                <circle style={{strokeDashoffset, stroke}} r="90" cx="100" cy="100" strokeDasharray="565.48" strokeDashoffset="0"></circle>
            </svg>
        </div>
    );
}

export default CirclePercent;
