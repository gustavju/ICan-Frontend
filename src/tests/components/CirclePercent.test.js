import React from 'react';
import { shallow } from 'enzyme';
import CirclePercent from '../../components/CirclePercent';

test('should render the CirclePercent', () => {
    const wrapper = shallow(<CirclePercent />);
    expect(wrapper).toMatchSnapshot();
});