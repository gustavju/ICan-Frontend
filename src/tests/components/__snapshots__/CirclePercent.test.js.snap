// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the CirclePercent 1`] = `
<div
  className="circle-percent__label"
>
  <svg
    className="circle-percent__svg"
    height="200"
    version="1.1"
    viewport="0 0 100 100"
    width="200"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="100"
      cy="100"
      r="90"
      strokeDasharray="565.48"
      strokeDashoffset="0"
    />
    <circle
      cx="100"
      cy="100"
      r="90"
      strokeDasharray="565.48"
      strokeDashoffset="0"
      style={
        Object {
          "stroke": undefined,
          "strokeDashoffset": NaN,
        }
      }
    />
  </svg>
</div>
`;
